dynet/devices.h

```diff
@@ -1,19 +1,16 @@
 #ifndef DYNET_DEVICES_H
 #define DYNET_DEVICES_H
 
+#define EIGEN_USE_THREADS
 #include <string>
+#include <unsupported/Eigen/CXX11/Tensor>
+
 #include "dynet/aligned-mem-pool.h"
 #include "dynet/cuda.h"
 
-namespace Eigen {
-  struct DefaultDevice;
-  class CudaStreamDevice;
-  struct GpuDevice;
-}
-
 namespace dynet {
 
-enum class DeviceType {CPU, GPU};
+enum class DeviceType {CPU, GPU, ThreadPool};
```

Comment:
You can understand ThreadPool as a device which relies on CPU but using multiple threads.